assets/RNAseq_report.Rmd

```diff
@@ -1486,16 +1494,84 @@ metadata <- metadata_PA
 metadata$Secondary.Name <- gsub(" ; ", "_", metadata$Secondary.Name) # Remove blank spaces and ; from secondary name
 metadata$Secondary.Name <- gsub(" ", "_", metadata$Secondary.Name) # Remove blank spaces if there from secondary name
 
-# Create output directory
-dir.create("pathway_analysis")
+#This saves whether a GMT file is present and should be moved to the pathway_analysis/metadata folder; initialize as FALSE
+mv_status <- F
+
+if (isProvided(params$custom_gmt)){
+
+    # If custom GMT file was provided, filter it before uploading to use in gost queries (gprofiler will NOT filter itself, so otherwise the output will contain all kinds of datasources, not just the selected ones)
+    out_gmt <- (Filter(function(line) any(startsWith(line, datasources)), readLines(params$custom_gmt)))
+    out_path <- paste0(tools::file_path_sans_ext(basename(params$custom_gmt)), "_filtered.gmt")
+    writeLines(out_gmt, out_path)
+    gost_id <- upload_GMT_file(out_path)
+    mv_status <- T
+} else {
+
+    # Otherwise, get the GMT file from gprofiler and save both the full file as well as the filtered one to metadata
+    gost_id <- organism
+    gmt_url <- paste0("https://biit.cs.ut.ee/gprofiler//static/gprofiler_full_", organism, ".ENSG.gmt")
+    tryCatch(
+        {
+            wget_command <- paste0("wget ", gmt_url)
+            sys_return <- system(wget_command)
+            if (sys_return == 0 || "saved" %in% sys_return) {
+                mv_status <- T
+            } else {
+                print("Failed to fetch the GMT file from gprofiler with this URL:")
+                print(gmt_url)
+                print("For reproducibility reasons, try to download the GMT file manually by visiting https://biit.cs.ut.ee/gprofiler/gost, then selecting the correct organism and finally, in datasources, clicking 'combined ENSG.gmt'.")
+            }
+        },
+        error=function(gost_error) {
+            print("Failed to fetch the GMT file from gprofiler with this URL:")
+            print(gmt_url)
+            print("Got error:")
+            print(gost_error)
+            print("For reproducibility reasons, try to download the GMT file manually by visiting https://biit.cs.ut.ee/gprofiler/gost, then selecting the correct organism and finally, in datasources, clicking 'combined ENSG.gmt'.")
```

Comment:
```suggestion
            print("For reproducibility reasons, try to download the GMT file manually by visiting https://biit.cs.ut.ee/gprofiler/gost, then selecting the correct organism and finally, in datasources, clicking 'combined ENSG.gmt'. Then provide it to the pipeline with the parameter `--custom_gmt`")
```

Reply:
Aaah, of course I forgot to update the docu! Just added the entry